tests/TurboMqtt.Tests/NonWindowsTheoryAttribute.cs

```diff
@@ -0,0 +1,84 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="NonWindowsTheoryAttribute.cs" company="Petabridge, LLC">
+//      Copyright (C) 2024 - 2024 Petabridge, LLC <https://petabridge.com>
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System.Runtime.InteropServices;
+
+namespace TurboMqtt.Tests;
+
+/// <summary>
+/// <para>
+/// This custom XUnit Fact attribute will skip unit tests if the run-time environment is windows
+/// </para>
+/// <para>
+/// Note that the original <see cref="Skip"/> property takes precedence over this attribute,
+/// any unit tests with <see cref="NonWindowsFactAttribute"/> with its <see cref="Skip"/> property
+/// set will always be skipped, regardless of the environment variable content.
+/// </para>
+/// </summary>
+public class NonWindowsFactAttribute : FactAttribute
```

Comment:
The SkippableFact package kept marking the tests as failed, which was driving my test coverage tool and me insane